bottomsheet/src/main/java/com/flipboard/bottomsheet/BottomSheetLayout.java

```diff
@@ -178,6 +181,35 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
         this.contentClipRect.set(0, 0, getWidth(), bottomClip);
     }
 
+    @Override
+    public boolean onKeyPreIme(int keyCode, KeyEvent event) {
+        if (keyCode == KeyEvent.KEYCODE_BACK && isSheetShowing()) {
+            if (event.getAction() == KeyEvent.ACTION_DOWN && event.getRepeatCount() == 0) {
+                KeyEvent.DispatcherState state = getKeyDispatcherState();
+                if (state != null) {
+                    state.startTracking(event, this);
+                }
+                return true;
+            } else if (event.getAction() == KeyEvent.ACTION_UP) {
+                KeyEvent.DispatcherState state = getKeyDispatcherState();
+                if (state != null) {
+                    state.handleUpEvent(event);
+                }
+                if (event.isTracking() && !event.isCanceled()) {
+                    if (isSheetShowing()) {
+                        if (getState() == BottomSheetLayout.State.EXPANDED) {
+                            peekSheet();
```

Comment:
I do want to bring this up again. In my case I'm using the bottom sheet for sharing. If you press back you want to completely cancel the share (what the default share sheet on lollipop does). If this is hard-coded to peek instead of dismiss, it will be vary hard to change.